crates/exex/src/model.rs

```diff
@@ -68,37 +94,43 @@ impl From<u64> for KethMaybeRelocatable {
 }
 
 /// [`KethOption`] is a custom representation of a Rust [`Option<T>`] type where `T` can be
-/// converted into [`KethMaybeRelocatable`], specifically tailored for use within the Keth model
-/// inside our OS.
+/// any type, such as [`KethMaybeRelocatable`] or [`KethU256`], specifically tailored for use
+/// within the Keth model inside our OS.
 ///
 /// This struct encapsulates two fields:
 /// - `is_some`: A flag that indicates whether the option contains a value (`is_some = 1`) or is
 ///   empty (`is_some = 0`).
-/// - `value`: The actual value of the option, represented as a [`KethMaybeRelocatable`].
+/// - `value`: The actual value of the option, which can be of any type `T`.
 ///
-/// When the option is [`None`], the `value` field is assigned a default zero value. This struct
+/// When the option is [`None`], the `value` field is assigned a default value. This struct
 /// provides a way to handle optional values in a manner consistent with the Keth model's memory
 /// representation.
 #[derive(Debug, Eq, Ord, Hash, PartialEq, PartialOrd, Clone, Serialize, Deserialize)]
-pub struct KethOption {
+pub struct KethOption<T> {
```

Comment:
as per our discussion, `is_some` could be a `Felt252` directly

Reply:
But in the end to fit into the `gen_arg`, `is_some`, despite being a `Felt` will have to be placed as a `MaybeRelocatable`. So maybe it's better to leave everything like that so as not to mix up things no? That way we can tell that everything has the same type.